lib/TPP/Dialect/Tpp/TppUtils.cpp

```diff
@@ -378,6 +378,8 @@ bool isTppRelu(linalg::GenericOp linalgOp) {
     return false;
   if (!isUnaryOp(linalgOp))
     return false;
+  if (!allOperandsHaveSameShape(linalgOp))
```

Comment:
Does this check that the stride is compatible with the iteration order across all operands?

Reply:
what do you mean?

Reply:
In only checks that the two memref have the same dimensions (i.e., memref<4x4xf32>)

Reply:
> what do you mean?

`memref<4x4xf2>` and `memref<4x4xf32, stride=[4, 1]>` are "compatible".

`memref<4x4xf2>` and `memref<4x4xf32, stride=[4, 2]>` are "not compatible". (jumps one column).

`memref<4x4xf2>` and `memref<4x4xf32, stride=[8, 1]>` are "not compatible". (jumps one line).